File: python/Happy Number.py
class Solution(object):
    def isHappy(self, n):
        tmp = n
        while tmp >= 10:
            tmp = 0
            while n:
                tmp += (n % 10) ** 2
                n //= 10
            n = tmp
        
        return tmp == 1 or tmp == 7

File: cpp/Happy Number.cpp
class Solution {
public:
    bool isHappy(int n) {
        do {
            int sum = 0;
            while (n > 0) {
                sum += (n%10) * (n%10);
                n /= 10;
            }
            n = sum;
        } while (n > 9);
        return n == 1;
    }
};
